crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -1,29 +1,32 @@
-use plugin_sm::log_file::LogFile;
-use std::cmp::Reverse;
 use std::collections::BinaryHeap;
 use std::path::PathBuf;
+use std::{cmp::Reverse, collections::HashMap};
 use time::{format_description, OffsetDateTime};
 use tracing::log;
 
+use crate::log_file::LogFile;
+
 #[derive(Debug, thiserror::Error)]
 pub enum OperationLogsError {
     #[error(transparent)]
     FromIo(#[from] std::io::Error),
 
     #[error(transparent)]
     FromTimeFormat(#[from] time::error::Format),
+
+    #[error("Operation name is None. Expected Some(`value`).")]
+    NoOperation,
 }
 
 #[derive(Debug)]
 pub struct OperationLogs {
-    log_dir: PathBuf,
-    max_update_logs: usize,
-    max_list_logs: usize,
+    pub log_dir: PathBuf,
 }
 
 pub enum LogKind {
     SoftwareUpdate,
     SoftwareList,
+    Operation,
```

Comment:
I get a feeling that this `LogKind` enum itself is unnecessary. It's just a thin wrapper over a `String`. Since this log-type is going to be an open string in the near future with #1077 , this enum will lose its relevance completely. So, my suggestion would be to get rid of this enum and just use some constants named `software-update` and `software-list` to represent those operation names.

It will also simplify the signature of the `pub async fn new_log_file( &self, kind: LogKind, operation_name: Option<&str>)` into just `pub async fn new_log_file( &self, operation_name: Option<&str>)`